internal/controller/dscinitialization/dscinitialization_controller.go

```diff
@@ -417,16 +441,20 @@ func (r *DSCInitializationReconciler) newMonitoringCR(ctx context.Context, dsci
 		},
 	}
 
+	// if metrics is not removed from DSCI totally, otherwise it is default as nil in defaultMonitoring
 	if dsci.Spec.Monitoring.Metrics != nil {
-		defaultMonitoring.Spec.Metrics = dsci.Spec.Monitoring.Metrics
+		// when metrics has values set in resoures or storage. skip replicas since it cannot be 0 from CEL validation
+		if dsci.Spec.Monitoring.Metrics.Storage != nil || dsci.Spec.Monitoring.Metrics.Resources != nil {
+			defaultMonitoring.Spec.Metrics = dsci.Spec.Monitoring.Metrics
+		} else { // if metrics is set to metrics:{} to avoid  invalid value "null" to patch existing Monitoring CR
```

Comment:
Now that we have replicas there, which are non-pointer type, SSA may not complain because even if we assign empty Metrics {} it will default to zero for the replica field so maybe now it will work?

Also now we will have slight discrepancy between DSCI and Monitoring - if dsci.Spec.Monitoring.Metrics will be Metrics {}, then in Monitoring the metrics will be nil/not present.